packages/app-builder-lib/src/node-module-collector/index.ts

```diff
@@ -0,0 +1,28 @@
+import { NpmNodeModulesCollector } from "./npmNodeModulesCollector"
+import { PnpmNodeModulesCollector } from "./pnpmNodeModulesCollector"
+import { YarnNodeModulesCollector } from "./yarnNodeModulesCollector"
+import { detect, PM, getNpmVersion } from "./packageManager"
+import { NodeModuleInfo } from "./types"
+import { log } from "builder-util"
+
+async function getCollectorByPackageManager(rootDir: string) {
+  const manager: PM = await detect({ cwd: rootDir })
+  switch (manager) {
+    case "npm":
+      return new NpmNodeModulesCollector(rootDir)
+    case "pnpm":
+      return new PnpmNodeModulesCollector(rootDir)
+    case "yarn":
+      return new YarnNodeModulesCollector(rootDir)
+    default:
+      log.warn({ rootDir }, `Cannot detect lock file to determine node_modules structure. Assuming plain structure.`)
```

Comment:
Please use `log.filePath` when logging file paths to reduce console log bloat.
e.g. `rootDir: log.filePath(rootDir)`